cli/src/commands/new.rs

```diff
@@ -206,6 +209,27 @@ Please use `jj new 'all:x|y'` instead of `jj new --allow-large-revsets x y`.",
     if num_rebased > 0 {
         writeln!(ui.stderr(), "Rebased {num_rebased} descendant commits")?;
     }
+
+    if !advanceable_branches.is_empty() {
+        tx.advance_branches(advanceable_branches, target_commits[0].id());
+    }
+
     tx.finish(ui, "new empty commit")?;
     Ok(())
 }
+
+// Branches are only advanced if `jj new` has a single target commit and the
+// new commit is not being inserted before or after the target.
+fn get_advanceable_branches(
+    args: &NewArgs,
+    ws: &WorkspaceCommandHelper,
+    target_commits: &[Commit],
+) -> Result<Vec<AdvanceableBranch>, CommandError> {
+    let should_advance_branches =
+        target_commits.len() == 1 && !args.insert_before && !args.insert_after;
+    if should_advance_branches {
+        return ws.get_advanceable_branches(target_commits[0].parent_ids());
+    } else {
+        Ok(Vec::new())
+    }
```

Comment:
nit: maybe simpler to just inline this function? The caller has to know that `target_commits` is singular if `advanceable_branches` is not empty.

Alternatively, the assumption can be encoded as `Option<(Vec<AdvanceableBranch>, CommitId)>`, but I'm not sure if that's better.

Reply:
Personally I'm not too worried about this since there's a comment explaining and the function is easy to read. I just wanted to pull the logic out of the main function to make it more readable. In general I think many of the command functions are a bit too long and would be easier to read if we pulled some of the details into functions.

If you want I'm fine inlining this, but I also think it's ok as-is.